### PR TITLE
Room persistence

### PR DESCRIPTION
**Pull request by christymarc**
Persistence of temporary timeline data

### CODE REVIEW

**Comment by christymarc** [comment]
Persistence of temporary feed

File: app/src/main/java/com/example/surfstop/ShortPostDetailActivity.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 import models.BasePost;
 import models.ShortPost;
+import utils.InternetUtil;
```
Comment (sefar) [inline app/src/main/java/com/example/surfstop/ShortPostDetailActivity.java:17]: Doesn't seem used.

File: app/src/main/java/fragments/DescriptionBoxFragment.java [hunk 2]
```diff
@@ -90,7 +91,11 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         spinnerBeach = view.findViewById(R.id.spinnerBeach);
         spinnerBeach.setOnItemSelectedListener(this);
 
-        QueryUtils.queryBeachesForSpinner(spinnerBeach, view);
+        if (InternetUtil.isInternetConnected()) {
+            QueryUtils.queryBeachesForSpinner(spinnerBeach, view);
+        } else {
+            QueryUtils.queryBeachesforSpinnerOffline(spinnerBeach, view);
+        }
```
Comment (sefar) [inline app/src/main/java/fragments/DescriptionBoxFragment.java:98]: Yeah, I'd move the Offline/not logic into the QueryUtils. Just have QueryUtils.queryBeachesForSpinner() { if (InternetUtil.isInternetConnected()) {
            QueryUtils.queryBeachesForSpinnerOnline(spinnerBeach, view);
        } else {
            QueryUtils.queryBeachesforSpinnerOffline(spinnerBeach, view);
        }
}

File: app/src/main/java/models/RoomBeachGroup.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+//        this.group = beachGroup.getKeyGroup();
+//        this.groupId = group.getObjectId();
+//        this.description = group.getKeyDescription();
+//        this.image = group.getKeyImage();
+//        this.imageUrl = image.getUrl();
+//        this.groupName = group.getKeyGroupName();
+//        this.maxBreak = beachGroup.getKeyMaxbreak();
+//        this.minBreak = beachGroup.getKeyMinbreak();
+//        this.waterTemp = beachGroup.getKeyWatertemp();
+//        this.locationId = beachGroup.getKeyLocationid();
```
Comment (sefar) [inline app/src/main/java/models/RoomBeachGroup.java:59]: ?

File: app/src/main/java/models/RoomGroup.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    public RoomGroup() { }
+
+    @Ignore
+    public RoomGroup(Group group) {
+        this.group = group;
+        this.groupObjectId = group.getObjectId();
+//        this.description = group.getKeyDescription();
+//        this.image = group.getKeyImage();
+//        this.imageUrl = image.getUrl();
+//        this.groupName = group.getKeyGroupName();
```
Comment (sefar) [inline app/src/main/java/models/RoomGroup.java:43]: ?

File: app/src/main/java/models/RoomShortPostWithObjects.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    @Embedded
+    RoomShortPost roomShortPost;
+
+    public static List<RoomShortPost> getRoomShortPostList(List<RoomShortPostWithObjects> postsWithObjects, BeachGroup beachGroup) {
+        List<RoomShortPost> posts = new ArrayList<>();
+        for (int i = 0; i < postsWithObjects.size(); i++) {
+            RoomShortPost post = postsWithObjects.get(i).roomShortPost;
+            post.roomUser = postsWithObjects.get(i).roomUser;
+            System.out.println(post.roomBeachGroupId);
+            System.out.println(beachGroup.getObjectId());
```
Comment (sefar) [inline app/src/main/java/models/RoomShortPostWithObjects.java:23]: Debug logging?

File: app/src/main/java/models/RoomShortPostWithObjects.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            users.add(user);
+        }
+        return users;
+    }
+
+    //    @Embedded (prefix = "group_")
+//    RoomGroup roomGroup;
+//
+//    @Embedded (prefix = "beach_")
+//    RoomBeachGroup roomBeachGroup;
```
Comment (sefar) [inline app/src/main/java/models/RoomShortPostWithObjects.java:44]: ?

File: app/src/main/java/models/ShortPost.java [hunk 7]
```diff
@@ -1,34 +1,44 @@
...
-    public String getKeyTag() {
-        return getString(KEY_TAG);
+    public ShortPost() { }
+
+    public ShortPost(RoomShortPost post, ParseUser user) {
+        this.setKeyContent(post.content);
+        this.setKeyUser(user);
+        this.setKeyCreatedAt(post.createdAt);
+        //this.setKeyGroup(post.group);
+        //this.setKeyBeachGroup(post.beachGroup);
```
Comment (sefar) [inline app/src/main/java/models/ShortPost.java:25]: ?

**Review by sefar — APPROVED** [review]

**Pull request merged**